sematic/cli/clean.py

```diff
@@ -0,0 +1,78 @@
+# Standard Library
+import logging
+import os
+from logging.config import dictConfig
+
+# Third-party
+import click
+
+# Sematic
+from sematic.api.endpoints.auth import get_cleaner_api_key
+
+# TODO: will use API client to perform cleaning operations
+# import sematic.api_client as api_client
+from sematic.cli.cli import cli
+from sematic.config.user_settings import UserSettingsVar
+from sematic.logging import make_log_config
+
+logger = logging.getLogger(__name__)
+
+
+@cli.command("clean", short_help="Clean up orphaned objects")
+@click.option(
+    "--orphaned-jobs",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help="Clean jobs whose associated runs have stopped.",
+)
+@click.option(
+    "--orphaned-resources",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help="Clean up external resources (ex: Ray clusters) whose runs have stopped.",
+)
+def clean(orphaned_jobs: bool, orphaned_resources: bool):
+    """
+    Clean up objects that are no longer needed.
+    """
+    running_as_cron_job = os.environ.get("RUNNING_AS_CRON_JOB", None) is not None
+    if running_as_cron_job:
+        dictConfig(make_log_config(log_to_disk=False))
+        echo = logger.info  # type: ignore
+        api_key = get_cleaner_api_key()
```

Comment:
I guess a longer term solution (way out of scope here) is to support authenticated apps, and have this cron job be a registered app.

Reply:
or have some more formal concept of Sematic service accounts